tests/utilities/grid_to_grid_projection.cc

```diff
@@ -0,0 +1,657 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2025 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// deal.II
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/distributed/fully_distributed_tria.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/fe/fe_raviart_thomas.h>
+#include <deal.II/fe/mapping_fe.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// ExaDG
+#include <exadg/grid/grid_data.h>
+#include <exadg/operators/finite_element.h>
+#include <exadg/operators/solution_projection_between_triangulations.h>
+#include <exadg/postprocessor/error_calculation.h>
+#include <exadg/postprocessor/write_output.h>
+#include <exadg/utilities/numbers.h>
+#include <exadg/operators/solution_projection_between_triangulations.cpp> // temporary
```

Comment:
I think you need to add this file in the CMakeLists file.

Reply:
ah, you did ... but the function in that file have many template parameters.

So I guess a .cpp file does not make sense in this form (without instantiations for all possible values of the template parameters). I suggest to put everything into the .h file.

Reply:
I would rather keep the separation in `.h` and `.cpp`.
Currently, we need the `.cpp` to be included like this, but we can remove it immediately after, when focusing on the restart procedures.